src/Firebase/AppCheck/AppCheckTokenOptions.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase\AppCheck;
+
+use InvalidArgumentException;
+use JsonSerializable;
+
+use function array_key_exists;
+
+/**
+ * @phpstan-type AppCheckTokenOptionsShape array{
+ *     ttl: string|null,
+ * }
+ */
+final class AppCheckTokenOptions implements JsonSerializable
+{
+    private ?string $ttl;
+
+    private function __construct(?string $ttl = null)
+    {
+        $this->ttl = $ttl;
+    }
+
+    /**
+     * @param array<string, mixed> $data
+     */
+    public static function fromArray(array $data): self
+    {
+        if (!array_key_exists('ttl', $data)) {
+            throw new InvalidArgumentException('The "app_id" key is missing from the token data.');
+        }
```

Comment:
`ttl` instead of `app_id` in the error message

I'm thinking about removing many of these runtime checks, not just here but in many places of the SDK where PHPStan/Psalm PHPDocs are in place.

For example, here, the `AppCheckTokenOptionsShape` already defines that a `ttl` key _must_ be present, so I'm wondering if runtime checks are even necessary.

So, unless you have strong opinions against it, I think it would be good to remove runtime checks where static analysis would already catch the problem.

(Strong Opinions against it: except "But most people don't use Static Analysis Tools" - because they should™ 😅)